File: tests/Tutorial/vectorisation-C/catenary.c
void catenary(double *y, const double *x, const double *p) {
	y[0] = x[1];
	y[1] = (1.0+x[1]*x[1])/x[0];
}

void catenary_v(const long int N, double *y, const double *x, const double *p) {
	long int i=0;
	for(i=0; i<N; i++, y+=2,x+=2) {
		y[0] = x[1];
		y[1] = (1.0+x[1]*x[1])/x[0];
	}
}

File: tests/Tutorial/Initial_Examples/Chapter_1/Alternative_3/catenary.m
function y = catenary(f, fp, p, mode)

switch mode
    case 'dLdf'
        y = sqrt(1+fp.^2);
    case 'dLdfp'
        y = f.*fp./sqrt(1+fp.^2);
    case 'd2Ldfdf'
        y = zeros(size(f));
    case 'd2Ldfpdf'
        y = fp./sqrt(1+fp.^2);
    case 'd2Ldfpdfp'
        y = f./(1+fp.^2).^(3/2);
end

end